uelc/main/models.py

```diff
@@ -66,11 +66,11 @@ def display_name(self):
         return '%s - %s' % (self.name)
 
     @classmethod
-    def create(self, cohort, hierarchy):
-        case = self(name=self.name,
-                    hierarchy=self.hierarchy,
-                    cohort=self.cohort)
-        return case
+    def create(self, request):
+        return Case.objects.create(
+            name=request.POST.get('name', ''),
+            hierarchy=request.POST.get('hierarchy', ''),
+            cohort=request.POST.get('cohort', ''))
 
```

Comment:
Maybe? As the code reads, the create could result in this scenario: Case.objects.create(name='', hierarchy='', cohort=''). Case is defined as name blank=False, so an exception will result. That's okay as long as the code downstream handles that scenario.

Also - just noting that cohort is a foreign key. I think cohort would need to be instantiated before being used in the create statement, yeah?